spring-integration-file/src/main/java/org/springframework/integration/file/support/FileUtils.java

```diff
@@ -29,6 +38,53 @@
 
 	public static final boolean IS_POSIX = FileSystems.getDefault().supportedFileAttributeViews().contains("posix");
 
+	/**
+	 * Remove entries from the array if the predicate returns true for an element.
+	 * @param fileArray the array.
+	 * @param predicate the predicate.
+	 * @param <F> the file type.
+	 * @return the array (modified or not), or null if all elements are removed.
+	 * @since 5.0.7
+	 */
+	public @Nullable static <F> F[] purgeUnwantedElements(F[] fileArray, Predicate<F> predicate) {
```

Comment:
Can't we do like this instead:
````java
@SuppressWarnings("unchecked")
	@Nullable
	public static <F> F[] purgeUnwantedElements(F[] fileArray, Predicate<F> predicate) {
		Object[] files =
				Arrays.stream(fileArray)
						.filter(predicate.negate())
						.toArray();

		if (files.length == 0) {
			return null;
		}
		else {
			return (F[]) files;
		}
	}
````

Or do you really worry about `Stream` instances all those support objects, plus empty array object if that?
However at the same time you do `new LinkedList<>(Arrays.asList(fileArray));`...

Reply:
Cool; pushing.

Reply:
Hm. OK. 
How about this then:
````java
return Arrays.stream(fileArray)
				.filter(predicate.negate())
				.toArray(size -> (F[]) Array.newInstance(fileArray[0].getClass(), size));
````
?